src/main/java/dao/FileReader.java

```diff
@@ -0,0 +1,8 @@
+package dao;
+
+import java.util.List;
+import model.FruitTransaction;
+
+public interface FileReader {
+    List<FruitTransaction> readFile(String fileName);
```

Comment:
The method signature of 'readFile' in the FileReader interface suggests that this FileReader is tightly coupled with the FruitTransaction model. According to the SOLID principles, and more specifically the Single Responsibility Principle, the FileReader should be generic and not tied to a specific model. Consider changing the return type to List<String> which will return the raw data lines from the file.